src/Avalonia.Base/Input/Gestures.cs

```diff
@@ -341,7 +343,7 @@ private static void PointerMoved(RoutedEventArgs ev)
 
                         if (s_isHolding)
                         {
-                            i.RaiseEvent(new HoldingRoutedEventArgs(HoldingState.Cancelled, s_lastPressPoint, s_lastHeldPointer!.Type, e));
```

Comment:
This solution is fine, though not perfect as you described. There shouldn't be situation, when s_lastHeldPointer is null but s_isHolding is true.

Possibly, we should create a "state struct":
```csharp
private struct GestureState(IPointer Pointer, Point Point, GestureStateType Type);
private enum GestureStateType
{
   Pending,
   Holding,
   DoubleTapped
}
```

And replace these local static fields with atomic single GestureState field, nullable. 
In this way, when one changes holding state, they have to specify full struct at once. And when they need to reset it - they nullify whole struct field.

Reply:
@maxkatz6 looks like here is a race condition. so state struct will not help, if I understand the things right.

```C#
                        if (_gestureState.Type == GestureStateType.Holding)
                        {
//other thread is changing _gestureState at this moment, so _gestureState.Pointer.Type means something wrong from now.
                            i.RaiseEvent(new HoldingRoutedEventArgs(HoldingState.Cancelled, Pointer.Point, 
_gestureState.Pointer.Type, e));
```

probably we will need to have a local copy of the struct. if `_gestureState` will be nullable the situation might be event worse...

PTAL, that this PR will not fix the race itself, it is only fix the bug when **NRE killed the entire app**. For us, it is a critical bug.

May be it will be better, to merge this PR as is, and create a follow up issue to track race and fix the race with approriate codechange? What do you think?

Reply:
I don't think there could be multiple threads involved, since Avalonia dispatcher is single threaded.

Reply:
Same with DispatcherTimer, it should be on the same thread.

Reply:
@maxkatz6 I did as you said, create GestureState struct.

Reply:
Unfortunately, i had to move `Point Point` field back to a separate `s_lastPressPoint` field due to failed tests. These tests call `PointerPressed` twice, and then also call `PointerReleased` twice. The second time `PointerReleased` is called, `s_gestureState` is `null`, but last point is still needed to handle this event. This cannot be done without `s_lastPressPoint` field.

Reply:
@maxkatz6 this is the fix for app crashing. would be nice if this fix will be delivered with next release of avalonia. sorry to bother.